File: Easy/Maximum Sum Problem/maximum-sum-problem.cpp
//{ Driver Code Starts
#include<bits/stdc++.h>
using namespace std;

// } Driver Code Ends
class Solution
{
    public:
        int maxSum(int n)
        {
            //code here.
            if(n == 0) return 0;
            int n1 = maxSum(n/2);
            int n2 = maxSum(n/3);
            int n3 = maxSum(n/4);
            return max(n, n1 + n2 + n3);
        }
};

//{ Driver Code Starts.
int main()
{
    int t;
    cin>>t;
    while(t--)
    {
        int n;
        cin>>n;
        Solution ob;
        cout<<ob.maxSum(n)<<"\n";
    }
    return 0;
}
// } Driver Code Ends

File: Easy/Maximum Sum Problem/maximum-sum-problem.java
//{ Driver Code Starts
//Initial Template for Java

import java.io.*;
import java.util.*;
class GfG
{
    public static void main(String args[])
        {
            Scanner sc = new Scanner(System.in);
            int t = sc.nextInt();
            while(t-->0)
                {
                    int n = sc.nextInt();
                    Solution ob = new Solution();
                    System.out.println(ob.maxSum(n));
                }
        }
}    
// } Driver Code Ends


//User function Template for Java

class Solution
{
    public int maxSum(int n) { 
        //code here.
        if(n == 0) return 0;
        int n1 = maxSum(n/2);
        int n2 = maxSum(n/3);
        int n3 = maxSum(n/4);
        return Math.max(n, n1 + n2 + n3);
    } 
}
